### DISCUSSION TITLE
RemoteGPIO Digital_Inputs & Relays numbering

### DISCUSSION THREAD

**Discussion by Lucifer06** [Lucifer06/RemoteGPIO#33] [Ideas]
Given the various platforms (rPI, Ekrano, Cerbo) that has from 0 up to 4 DI and 0 or 2 Relays, we need to discuss if we always want to have no discontinuity with the numbering or the I/Os.

We need to decide what is better:
1/ RemoteGPIO will always start with the same numbering regardless of host platform.
Simple this is working like this today.

2/ Depending of the host platform, the RemoteGPIO will start numbering such a way that there will be no gaps.
It will require some more work...

What do you think?

**Comment by drtinaz** [comment]
Given the method that S90rgpio_pins.sh uses to find the first number to use for the relays and digital inputs when installing the links and creating conf files, it shouldn't be too difficult to check what platform is being used, and adjust the starting number accordingly.

**Comment by kwindrem** [comment]
I seem to remember that gaps in the digital input or relay numbering caused issues somewhere.

(Moved from another discussion which is now closed

**Comment by Albarge** [comment]
Hi, If the numbering can be assigned based on platform, does that mean it may be possible for a rPI to have relay 1 as generator start/stop?

**Reply by drtinaz** [reply]
This is not currently implemented but has been discussed for a future update.

**Reply by Albarge** [reply]
That would be great!
Especially when all digital in's are working, It'll make my transition from Cerbo over to a rPI pretty much fully functional in a simple way.
Thanks all for for working on this.
